roda-ui/roda-wui/src/main/java/org/roda/wui/client/ingest/process/PluginParameterPanel.java

```diff
@@ -313,6 +317,53 @@ public void onClick(ClickEvent event) {
     removeButton.addStyleName("toolbarLink toolbarLinkSmall");
   }
 
+  /**
+   * Method for creating filterParams for searching aips to dispose: disposaldate has to be prior today and
+   * disposalstatus has to be set in ClientMessages.properties to a relevant value.
+   * dateFinal and preservationstatus_txt are specified search fields in Swedish Customs gip.xml
+   * @return List of filterParams
+   */
+  private static List<FilterParameter> getFilterParameters(){
```

Comment:
This whole method seems very specific to a certain execution. Start date should not be fixed to 1970-01-01, search should not be fixed to a field 'preservationstatus_txt' that is not part of the base configuration. Plugin configuration should not be put into messages, like messages.disposalStatus(), but from RodaCoreFactory.getConfiguration() which receives the plugin specific configuration on plugin load. 

In global, this whole method should be replaced by a method that gets a filter for a configurable plugin parameter using the plugin configuration.

Reply:
How do I get access to the RodaCoreFactory from PluginParameterPanel?

Reply:
Although there is a way to do this via the new [ConfigurationManager](https://github.com/keeps/roda/blob/master/roda-ui/roda-wui/src/main/java/org/roda/wui/common/client/tools/ConfigurationManager.java) my idea here was not do it that way. A plugin parameter, that comes from the plugin (server side), should have all needed information to construct the interface needed to configure the plugin. This would mean that it would be the Plugin that goes to its configuration (via the RodaCoreFactory to allow re-configuration) and defined the PluginParameter with the necessary information (button message, filter, etc.), than the interface should cope with this. This might mean that besides the PluginParameterType we need extra information to construct the interface.

Reply:
To be clear, what I suggest is to add a new attribute Optional<Serializable> renderingHints to the [PluginParameter], include getter and setter for this attribute. The plugin could define a plugin parameter of type AIP_ID and with renderingHints with the required filter (or more information if needed). Then update the PluginParameterPanel#createSelectAipLayout() to use this Filter, and further customize it with information that would come from renderingHints if needed.